pylot/perception/base_perception_eval.py

```diff
@@ -0,0 +1,194 @@
+                self._start_time_best_inference = p_start_time
+                self._prediction_start_end_times = \
+                    self._prediction_start_end_times[index:]
+                self._start_time_frontier -= index
+                break
+            index -= 1
+        if self._start_time_best_inference:
+            # Compute accuracy if we have a prediction with end_time less than
+            # current frontier.
+            self._accuracy_compute_buffer.append(
+                (self._start_time_best_inference, st, True))
+        self.__drain_accuracy_compute_buffer(st)
+        # Increase the frontier to process the next start time.
+        self._start_time_frontier += 1
+
+    def __drain_accuracy_compute_buffer(self, up_to_time):
+        for (st, et, end_anchored) in self._accuracy_compute_buffer:
+            if et <= up_to_time:
+                self.compute_accuracy(st, et, end_anchored)
+        self._accuracy_compute_buffer = list(
+            filter(lambda args: args[1] > up_to_time,
+                   self._accuracy_compute_buffer))
+        # Can safely GC until the minimum between the min start time in the
+        # accuracy compute buffer and the best inference start time.
+        gc_threshold = min(self._accuracy_compute_buffer, default=[None])[0]
+        if (self._start_time_best_inference is not None
+                and (gc_threshold is None
+                     or gc_threshold > self._start_time_best_inference)):
+            gc_threshold = self._start_time_best_inference
+        if gc_threshold is not None:
+            self.__gc_obstacles_earlier_than(gc_threshold)
+
+    def compute_accuracy(self, frame_time, ground_time, end_anchored):
+        raise NotImplementedError("To be implemented by child class.")
+
+    def __compute_frame_gap(self, game_time):
+        """Infer frame gap if not explicitly provided in constructor."""
+        if not self._frame_gap:
+            if not self._last_notification:
+                self._last_notification = game_time
+                return
+            else:
+                self._frame_gap = (game_time - self._last_notification)
+                self._last_notification = game_time
+
+    def get_ground_truth_at(self, timestamp):
+        for (ground_time, obstacles) in self._ground_truths:
+            if ground_time == timestamp:
+                return obstacles
+            elif ground_time > timestamp:
+                break
+        self._logger.fatal(
+            'Could not find ground obstacles for {}'.format(timestamp))
+
+    def get_prediction_at(self, timestamp):
+        for (start_time, obstacles) in self._predictions:
+            if start_time == timestamp:
+                return obstacles
+            elif start_time > timestamp:
+                break
+        self._logger.fatal(
+            'Could not find tracked obstacles for {}'.format(timestamp))
+
+    def __gc_obstacles_earlier_than(self, game_time):
```

Comment:
s/gc_obstacles/gc_predictions

Reply:
this function GCs both predictions and ground truth, so I'll switch to `__gc_data_earlier_than`